### ISSUE TITLE
Request static file with query return 404

### ISSUE THREAD

**Issue by ZXZhuang** [nguyenbatranvan/vite-multiple-assets#11]
Environment:
Windows 11; Vite 4.5.1;

Request static file with query return 404
```
const f = 'iconfont.woff2'
const file = 'iconfont.woff2?t=1702609282724'
path.relative(f, file) // => "..\\iconfont.woff2?t=1702609282724"
path.relative(f, file) === "" // false
```

**Comment by nguyenbatranvan** [comment]
> Environment: Windows 11; Vite 4.5.1;
> 
> Request static file with query return 404
> 
> ```
> const f = 'iconfont.woff2'
> const file = 'iconfont.woff2?t=1702609282724'
> path.relative(f, file) // => "..\\iconfont.woff2?t=1702609282724"
> path.relative(f, file) === "" // false
> ```

Maybe windows has a different path than macos =))

**Comment by ZXZhuang** [comment]
![image](https://github.com/nguyenbatranvan/vite-multiple-assets/assets/43693464/bbe08769-f784-499b-a285-d72972d6cff6)
![image](https://github.com/nguyenbatranvan/vite-multiple-assets/assets/43693464/f57cef5f-4d5d-4dbc-8361-e4feb0272649)
viteServePublicMiddleware clean request url to get correct file url

**Comment by nguyenbatranvan** [comment]
> ![image](https://private-user-images.githubusercontent.com/43693464/290724048-bbe08769-f784-499b-a285-d72972d6cff6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3MTU2NjMsIm5iZiI6MTcwMjcxNTM2MywicGF0aCI6Ii80MzY5MzQ2NC8yOTA3MjQwNDgtYmJlMDg3NjktZjc4NC00OTliLWEyODUtZDcyOTcyZDZjZmY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE2VDA4MjkyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2YzEzNmQyZjY5ZGZiYWRiODQ2YWQwZDI4NmQwYzBjY2I2MzljM2M2MzVkNjFjODcyNDIwMzdkZTgxNTFlMzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YNwi4PNBqxNQjcRpSD9Hl9D771mRDwxv3pqOdC30jdI) ![image](https://private-user-images.githubusercontent.com/43693464/290724069-f57cef5f-4d5d-4dbc-8361-e4feb0272649.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3MTU2NjMsIm5iZiI6MTcwMjcxNTM2MywicGF0aCI6Ii80MzY5MzQ2NC8yOTA3MjQwNjktZjU3Y2VmNWYtNGQ1ZC00ZGJjLTgzNjEtZTRmZWIwMjcyNjQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE2VDA4MjkyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4M2RhMTI3OTY4MmIyNmVmNTZiNjc3YjUyOTkzNjIzYjEwNWFjMWM0NmJkNzllOWM0ZmMzNzU0NGJmMmUxNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e4jFpAucOl-_4uSX7wwVostJmAfkXiKa-kQc5u3Zkzw) viteServePublicMiddleware clean request url to get correct file url

Hello! This is the solution you want to suggest me?

**Comment by ZXZhuang** [comment]
Yes.

**Comment by nguyenbatranvan** [comment]
> Yes.

Tks so much!! You can be a contributor if you want!!! By forking the project & creating a pull request for this issue

**Comment by nguyenbatranvan** [comment]
I have released version `1.2.10` about this issue, you can check and close it
